gateway/api/views.py

```diff
@@ -358,6 +358,19 @@ def _get_program_queryset_for_title_and_provider(
 
         return result_queryset
 
+    @action(methods=["GET"], detail=True)
+    def get_jobs(
+        self, request, pk=None
+    ):  # pylint: disable=invalid-name,unused-argument
+        """Returns jobs of the program."""
+        tracer = trace.get_tracer("gateway.tracer")
+        ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
+        with tracer.start_as_current_span("gateway.program.get_jobs", context=ctx):
+            program = Program.objects.filter(id=pk).first()
```

Comment:
We cannot return jobs only filtering by `id` because everyone could get `jobs` if it has that `id`. My recommendation for this would be:
- If the user is a provider returns all the jobs
- if the user is not a provider return **only** author jobs
- Let's use the serializer `JobSerializer` to return more info about the `Job`: 
  - https://github.com/Qiskit/qiskit-serverless/blob/main/gateway/api/v1/serializers.py#L102